### ISSUE TITLE
serveStatic w/ compress middleware issue

### ISSUE THREAD

**Issue by luwes** [honojs/node-server#104]
In my testing the compress middleware made a `serveStatic` HTML page return plain text `text/plain;charset=UTF-8`.
https://hono.dev/middleware/builtin/compress

**Comment by sebastianwessel** [comment]
I can confirm, that there is an issue with the response content-type header, as soon as compression is enabled.

Minimal example:

```typescript
import { serve } from '@hono/node-server'
import { Hono } from 'hono'
import { compress } from 'hono/compress'

const app = new Hono()

app.use('*', compress())

app.get('/', (c) =>
  c.json({
    some: 'value',
  }),
)

serve({
  fetch: app.fetch,
  port: 3000,
})
```

response: 
content-encoding: gzip
content-type: text/plain;charset=UTF-8

Tried it with node 20.8 and 20.10

The issue seems not to be present when with Deno.
As Bun isn't supporting compress right now, I couldn't test it.

I had a quick look into compress middleware and I didn't recognize something, that might be the root cause.
So, I think the issue is somewhere in the node-server package

**Comment by yusukebe** [comment]
@luwes @sebastianwessel Thank you for raising the issue.

Hi @usualoma, in the fowllowing case, test will fail. Can you see it?

```ts
describe('content-type', () => {
  const app = new Hono()
  app.use('*', async (c, next) => {
    await next()
    c.res = new Response('', c.res)
    c.res.headers // If this is set, test fails
  })

  app.get('/json', async (c) => {
    return c.json({})
  })

  it('Should return 200 response - GET /json', async () => {
    const server = createAdaptorServer(app)
    const res = await request(server).get('/json')
    expect(res.status).toBe(200)
    expect(res.headers['content-type']).toMatch(/application\/json/)
  })
})
```

**Comment by usualoma** [comment]
It appears to be a bug in the Response object. It can be fixed. Please wait a moment.

**Comment by yusukebe** [comment]
@usualoma 

Thanks!

**Comment by yusukebe** [comment]
Hi @luwes 

Thanks to @usualoma , this issue is fixed in latest version v1.3.2. Give it a try!

**Comment by sebastianwessel** [comment]
I did a quick test, and it seems to work as expected.
Thanks for the quick fix 🙏.

**Comment by yusukebe** [comment]
@luwes 

It's fixed thanks to @usualoma and I've released new version `v1.3.3` which includes the fix. Take a look!
